### ISSUE TITLE
[Bug]: Test doesn't fail as expected

### ISSUE THREAD

**Issue by beksajt** [microsoft/playwright-dotnet#2999]
### Version

1.46.0

### Steps to reproduce

```
await Page.GotoAsync("http://www.google.com/");
await Expect(Page).ToHaveTitleAsync("xD");
```

### Expected behavior
![pw 1 39 0](https://github.com/user-attachments/assets/67d836e2-7bc2-4165-82f1-8f7246ece33f)

Test result should be failed with exception:
`Microsoft.Playwright.PlaywrightException : Page title expected to be 'xD'`

### Actual behavior

Test can't finish its job and spins forever:
![pw 1 46 0](https://github.com/user-attachments/assets/dedec486-1ba6-4c66-bc50-a24a5eb279fb)

### Additional context

Found out that it happens only when chromium is set in runsettings. Firefox works ok.

### Environment

```Text
- Operating System: Windows10
- CPU: [arm64]
- Browser: [ Chromium ]
- .NET Version (TFM): [net8.0]
- Other info:
runsettings:
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
  <!-- NUnit adapter -->  
  <NUnit>
    <NumberOfTestWorkers>24</NumberOfTestWorkers>
  </NUnit>
  <!-- General run configuration -->
  <RunConfiguration>
    <EnvironmentVariables>
      <!-- For debugging selectors, it's recommend to set the following environment variable -->
      <DEBUG>pw:api</DEBUG>
    </EnvironmentVariables>
  </RunConfiguration>
  <!-- Playwright -->  
  <Playwright>
    <BrowserName>chromium</BrowserName>
    <ExpectTimeout>5000</ExpectTimeout>
    <LaunchOptions>
      <Headless>false</Headless>
      <Channel>msedge</Channel>
    </LaunchOptions>
  </Playwright>
</RunSettings>
```

**Comment by mxschmitt** [comment]
The logs you shared are still using Edge - do you have logs with just Chromium (no channel)?

**Comment by beksajt** [comment]
There you go
[no-channel-1.46.0.txt](https://github.com/user-attachments/files/16943878/no-channel-1.46.0.txt)

**Comment by mxschmitt** [comment]
Looking at the logs I observe that the directory cleanup seems to hang and then races with the 30s timer to kill the browser:

> 2024-09-10T11:07:19.901Z pw:browser [pid=9268] starting temporary directories cleanup
> 2024-09-10T11:07:49.188Z pw:browser [pid=9268] <kill>

It internally will retry when `EBUSY`, `EMFILE`, `ENFILE`, `ENOTEMPTY`, or `EPERM` error is encountered during directory removal.

Do you have a special anti virus installed on your system? Enterprise policies should not play a role when you use just Chromium.

**Comment by beksajt** [comment]
Nope, i dont use any special anti virus. When i downgrade nuget package to 1.39.0 test is finished with fail (as expected). Take a look a this log. There is an exception at the end, but at least it exits.
[no-channel-1.39.0.txt](https://github.com/user-attachments/files/16944220/no-channel-1.39.0.txt)

**Comment by mxschmitt** [comment]
I see, this is probably also the case except that we don't surface it anymore:

> 2024-09-10T11:03:35.922Z pw:browser [pid=14916] exception while removing C:\Users\$USER\AppData\Local\Temp\playwright_chromiumdev_profile-XXXXXXnNz44M: Error: EPERM: operation not permitted, unlink 'C:\Users\Konrad\AppData\Local\Temp\playwright_chromiumdev_profile-XXXXXXnNz44M\Default\Network\Cookies-journal'

[Looks like](https://stackoverflow.com/a/61210717/6512681) Chromium keeps the file open / in busy state.
